Add 500 character limit to description field.

Oceas opened the pull request:
Ticket: https://webdevstudios.atlassian.net/browse/CC-254

richaber commented on includes/class-builder-fields.php:
```diff
@@ -690,6 +690,7 @@ public function fields_metabox() {
 			'id'         => $this->prefix . 'field_desc',
 			'type'       => 'text',
 			'attributes' => [
+				'maxlength' => 500,
```
@Oceas I'm not sure how this resolves the ticket?

There's already a maxlength of 500 on the textarea, as noted in the ticket description, which can be found here https://github.com/WebDevStudios/constant-contact-forms/blob/fd3ae5b8483df8cf381efd91e76fef8d36b5c801/includes/class-display.php#L723-L724

The ticket says to either add a character counter to the textarea (via JS) or "perhaps just a note indicating there’s a 500 character limit", this PR does not appear to do either.

tw2113 replied:
Correct, this is for user-facing when visiting the site, not for admin side when building a form.

Oceas commented:
A label was added alerting users to 500 character limit has been added. 

Note originally (pre-pr) a user could actually save 500+ in the description field successfully. The purpose of the attribute makes CMB2 enforce a 500 character limit preventing users in the future from saving a 500+ character description. I did test to ensure that if a user has previously saved a 500+ character string it would be in place, but future descriptions are forced to be below 500.

tw2113 commented:
Scott, sorry for not being as clear as should be. This isn't for the admin area with the site owner, this is meant for the frontend when users are filling out the form. For example: https://cloudup.com/cr8JUqU9jaz where I added two textarea fields.

The markup for this is already similar like: `<textarea class="..." name="..." id="..." placeholder="" maxlength="500"></textarea>` with the maxlength attribute existing, but we don't mention that visually for the visitor.

Oceas commented:
Removed admin descriptions. Added 500 character limit with a custom css selector in the event a site owner wishes to style/hide the field. Ideally wanted a form setting however how the current display logic for rendering this field is setup would require a much larger refactor without much benefit.

richaber commented on includes/class-display.php:
```diff
@@ -1791,9 +1791,10 @@ public function textarea( $name = '', $map = '', $value = '', $desc = '', $req =
+		$return            = '<p class="' . implode( ' ', $classes ) . '">';
+		$label             = '<span class="' . $label_placement_class . '"><label for="' . esc_attr( $field_id ) . '">' . esc_attr( $name ) . ' ' . $req_label . '</label></span>';
+		$textarea          = '<textarea class="' . esc_attr( implode( ' ', $textarea_classes ) ) . '" ' . $req_text . ' name="' . esc_attr( $map ) . '" id="' . esc_attr( $field_id ) . '" placeholder="' . esc_attr( $desc ) . '" ' . $extra_attrs . '>' . esc_html( $value ) . '</textarea>';
+		$instructions_span = '<span class="ctct-textarea-warning-label"><label for="' . esc_attr( $field_id ) . '">' . esc_html__( 'Limit 500 Characters', 'constant-contact-forms' )  . ' ' . $req_label . '</label></span>';
```
```
phpcs: Squiz.Strings.ConcatenationSpacing.PaddingFound: Concat operator must be surrounded by a single space
```

richaber reviewed (approved):
It's a little hard to read, but so were the existing lines above it.

richaber commented:
Of course _now_ I notice that there's a <label> inside the span that shouldn't be there.

```
<p class="ctct-form-field">
    <span class="ctct-label-top">
        <label for="custom_text_area___99a6a1bc9800303822f0b441fd116a95_0">
            Custom Text Area 
        </label>
    </span>
    <textarea class="ctct-textarea ctct-label-top" name="custom_text_area___99a6a1bc9800303822f0b441fd116a95"
              id="custom_text_area___99a6a1bc9800303822f0b441fd116a95_0" placeholder="" maxlength="500">
    </textarea>
    <span class="ctct-textarea-warning-label">
        <label for="custom_text_area___99a6a1bc9800303822f0b441fd116a95_0">
            Limit 500 Characters 
        </label>
    </span>
</p>
```
